Tests/CommandsTests/RunToolTests.swift

```diff
@@ -0,0 +1,103 @@
+            let runOutput = try execute([], chdir: path)
+            let outputLines = runOutput.split(separator: "\n")
+            XCTAssertEqual(outputLines.last!, "[]")
+        }
+    }
+    
+    func testExplicitExecutable() throws {
+        fixture(name: "Miscellaneous/EchoExecutable") { path in
+            let runOutput = try execute(["secho"], chdir: path)
+            let outputLines = runOutput.split(separator: "\n")
+            XCTAssertEqual(outputLines.last!, "[]")
+        }
+    }
+    
+    func testArgumentPassing() throws {
+        fixture(name: "Miscellaneous/EchoExecutable") { path in
+            let runOutput = try execute(["secho", "1", "--hello", "world"], chdir: path)
+            let outputLines = runOutput.split(separator: "\n")
+            XCTAssertEqual(outputLines.last!, "[\"1\",\"--hello\",\"world\"]")
+        }
+    }
+    
+    func testNoExecutable() throws {
+        fixture(name: "Products/StaticLibrary") { path in
+            do {
+                _ = try execute([], chdir: path)
+                XCTFail("Unexpected success")
+            } catch SwiftPMProductError.executionFailure(_, _, let stderr) {
+                XCTAssertEqual(stderr, "error: no executable product in the package\n")
+            }
+        }
+    }
+    
+    func testUnkownExecutable() throws {
+        fixture(name: "Miscellaneous/EchoExecutable") { path in
+            do {
+                _ = try execute(["unknown"], chdir: path)
+                XCTFail("Unexpected success")
+            } catch SwiftPMProductError.executionFailure(_, _, let stderr) {
+                XCTAssertEqual(stderr, "error: could not find executable product 'unknown' in the package\n")
+            }
+        }
+    }
+    
+    func testMultipleProducts() throws {
+        fixture(name: "Miscellaneous/MultipleExecutables") { path in
+            do {
+                _ = try execute([], chdir: path)
+                XCTFail("Unexpected success")
+            } catch SwiftPMProductError.executionFailure(_, _, let stderr) {
+                XCTAssertEqual(stderr, "error: multiple executable products in the package. Use `swift run ` followed by one of: exec1, exec2\n")
+            }
+            
+            var runOutput = try execute(["exec1"], chdir: path)
+            var outputLines = runOutput.split(separator: "\n")
+            XCTAssertEqual(outputLines.last!, "1")
+            runOutput = try execute(["exec2"], chdir: path)
+            outputLines = runOutput.split(separator: "\n")
+            XCTAssertEqual(outputLines.last!, "2")
+        }
+    }
+
+    static var allTests = [
+        ("testUsage", testUsage),
+        ("testVersion", testVersion),
+        ("testImplicitExecutable", testImplicitExecutable),
+        ("testExplicitExecutable", testExplicitExecutable),
+        ("testArgumentPassing", testArgumentPassing),
+        ("testNoExecutable", testNoExecutable),
+        ("testUnkownExecutable", testUnkownExecutable)
```

Comment:
What is the (time) cost of all these functional tests? We should probably just have one functional test for completeness and cover other cases using unit tests.